projects/atlendis-v2/index.js

```diff
@@ -1,57 +1,90 @@
+const { GraphQLClient } = require("graphql-request");
 const query = `
-  query tvl($chainId: Int!) {
+  query tvlAndBorrow($chainId: Int!) {
     rcls(chainId: $chainId) {
-      id
-     address
+      address
       tvl
       token {
         address
       }
-      currentLoan {accruedInterests borrowedAmount }
+      currentLoan {
+        borrowedAmount
+      }
     }
     bulletLoans(chainId: $chainId) {
-      id
-     address
+      address
       tvl
       token {
         address
       }
+      issuedLoan {
+        borrowedAmount
+      }
     }
   }
-`
+`;
 
 // Atlendis' indexer url
 const atlendisUrl = "https://atlendis.herokuapp.com/graphql";
 // Atlendis V2 currently supports two chains: Polygon and Mode
 const supportedChains = { polygon: 137, mode: 34443 };
 
 async function tvl(api) {
-  const chain = api.chain
+  const chain = api.chain;
   const graphQLClient = new GraphQLClient(atlendisUrl);
   const chainId = supportedChains[chain];
-  const { rcls, } = await graphQLClient.request(query, { chainId });
-  const tokensAndOwners = rcls.map(rcl => [rcl.token.address, rcl.address])
-  return api.sumTokens({ tokensAndOwners })
+  const { rcls, bulletLoans } = await graphQLClient.request(query, { chainId });
+  const balances = {};
+  for (let rcl of rcls) {
+    const poolTvl = rcl.tvl - rcl.currentLoan?.borrowedAmount || "0x00";
+    if (!balances[`${chain}:${rcl.token.address}`]) {
+      balances[`${chain}:${rcl.token.address}`] = poolTvl;
+    } else {
+      balances[`${chain}:${rcl.token.address}`] =
+        balances[`${chain}:${rcl.token.address}`] + poolTvl;
+    }
+  }
+  for (let loan of bulletLoans) {
+    const poolTvl = loan.tvl - loan.issuedLoan?.borrowedAmount || "0x00";
+    if (!balances[`${chain}:${loan.token.address}`]) {
+      balances[`${chain}:${loan.token.address}`] = poolTvl;
+    } else {
+      balances[`${chain}:${loan.token.address}`] =
+        balances[`${chain}:${loan.token.address}`] + poolTvl;
+    }
+  }
+  return balances;
 }
 
 async function borrowed(api) {
-  const chain = api.chain
+  const chain = api.chain;
   const graphQLClient = new GraphQLClient(atlendisUrl);
   const chainId = supportedChains[chain];
   const { rcls, bulletLoans } = await graphQLClient.request(query, { chainId });
-
+  const balances = {};
   for (let rcl of rcls) {
     if (rcl.currentLoan) {
-      api.add(rcl.token.address, rcl.currentLoan.borrowedAmount)
+      const { borrowedAmount } = rcl.currentLoan;
+      if (!balances[`${chain}:${rcl.token.address}`]) {
+        balances[`${chain}:${rcl.token.address}`] = borrowedAmount;
+      } else {
+        balances[`${chain}:${rcl.token.address}`] =
+          balances[`${chain}:${rcl.token.address}`] + borrowedAmount;
+      }
     }
   }
-
   for (let loan of bulletLoans) {
     if (loan.issuedLoan) {
-      api.add(loan.token.address, loan.tvl)
+      const { borrowedAmount } = loan.issuedLoan;
+      if (!balances[`${chain}:${loan.token.address}`]) {
```

Comment:
can you skip the `balances` object, and use `api.add(loan.token.address, loan.issedLoan.borrowedAmount)` ?  shorter and easier to read

Reply:
ok for me to do so, the only thing is we expect at some point to have multiple pools with the same token to be deposited in, e.g 2 or 3 "USDC" pools. With the suggested implementation `api.add(loan.token.address, loan.issedLoan.borrowedAmount)`, only the last balance will be exposed, erasing the others. So i will need to sum the amounts of all pools before passing using `api.add(...)`

Reply:
no, the api.add takes care of it, if it was previously {USDC: 1000}, then you do api.add(USDC, 1000), it will now be {USDC:2000} it doesnt replace :)

Reply:
indeed ! my bad, the code i was testing was 'buggy'
ok, nice one then, publishing the changes in a few !
many thanks for your prompt answers